src/axom/primal/operators/detail/clip_impl.hpp

```diff
@@ -761,7 +755,22 @@ AXOM_HOST_DEVICE Polygon<T, 2, ARRAY_TYPE, MAX_VERTS> clipPolygonPolygon(
     }
   }  // end of iteration through edges of clip polygon
 
-  return outputList;
+  // Remove duplicate points.
+  // Handles edge cases such as 3 consecutive vertices with orientations
+  // ON_POSITIVE_SIDE, ON_BOUNDARY, ON_POSITIVE where point on the boundary
+  // is added twice.
+  PolygonType uniqueList;
+  for(int i = 0; i < outputList.numVertices(); i++)
+  {
+    int prevIndex = ((i - 1) == -1) ? (outputList.numVertices() - 1) : (i - 1);
+    if(outputList[i] == outputList[prevIndex])
```

Comment:
I think I would change this to a != comparison, remove the continue, and add the vertex inside this conditional.